### PR TITLE
[WIP]DHT11 support added

### LINKED ISSUE #1172: Implement DHT11 Temperature/Humidity sensor
**Actual Behaviour**

Temperature/Humidity sensor not available in PSLab Android

**Expected Behaviour**

Implement DHT11 sensor. Have a look at Arduino library for [DHT11](https://github.com/adafruit/DHT-sensor-library) and take inspiration from that.

**Steps to reproduce it**

> Not available

**LogCat for the issue**

> Not available

**Screenshots of the issue**

> Not available

**Would you like to work on the issue?**
@Vikum94

### PR DESCRIPTION
**Pull request by ho-dor**
Fixes #1172 

**Changes**: Adding support for DHT11

**Screenshot/s for the changes**: N/A

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**:N/A

### CODE REVIEW

**Comment by neel1998** [comment]
@ho-dor  i don't think 0x0f is the correct I2C address for DHT11 sensor. I checked on this link where address for all I2C sensors are specified https://learn.adafruit.com/i2c-addresses/the-list . @CloudyPadmal please guide. Also some of the sensors in #1840 are not I2C. How to communicate with such devices?

**Comment by ho-dor** [comment]
@neel1998 what about DHT11, is it i2c? It's not in the list you have mentioned

**Comment by neel1998** [comment]
@ho-dor  i am not sure. But mostly all the i2c sensors have SDA and SCL pins. Which are not there in DHT11. Still we have to check once. But mostly it might not be i2c

**Comment by CloudyPadmal** [comment]
DHT11 is not an I2C sensor. Checkout Arduino library and see how it's implemented.

File: app/src/main/java/io/pslab/communication/sensors/DHT11.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    public int NUMPLOTS = 1;
+    public String[] PLOTNAMES = {"Data"};
+    public String name = "Humidity/Temperature";
+
+    public ArrayList<String> selectParameter = new ArrayList<>(Arrays.asList("temperature", "humidity"));
+    private I2C i2c;
+
+    public DHT11(I2C i2c, ScienceLab scienceLab) throws IOException, InterruptedException {
+        this.i2c = i2c;
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/communication/sensors/DHT11.java]: Not I2C

**Comment by CloudyPadmal** [comment]
I suggest we close this for now and work on it later. DHT11 sensor works in a different principle and I think a direct firmware support would be more appropriate than trying hard with Android. I'll follow up with this issue later and closing this for now

**Pull request closed without merging**